### PR TITLE
allow ssh key option

### LINKED ISSUE #9: Can this be made to authenticate via ssh_key file?
netmiko appears to support this via CiscoSSHConnection

but mdssdk classes (eg Switch) don't have it in their __init__

I think this just needs to be passed down to the SSHSession.__init__._cisco_device dict, as its _connect method passes all the items to netmiko's ConnectHandler, yeah?

thanks!

### PR DESCRIPTION
**Pull request by wardy3**
- [x] allow ssh key file to be provided instead of password
- [ ] write tests for passwordless access

closes #9

### CODE REVIEW

**Comment by srbharadwaj** [comment]
Can you please add an example for the same?

File: mdssdk/switch.py [hunk 1]
```diff
@@ -54,14 +55,17 @@ class Switch(SwitchUtils):
 
     :example:
         >>> switch_obj = Switch(ip_address = switch_ip, username = switch_username, password = switch_password)
+        >>> switch_obj = Switch(ip_address = switch_ip, username = switch_username, use_keys = True, key_file = './ssh/test_rsa')
```
Comment (srbharadwaj) [inline mdssdk/switch.py:58]: Why do we need `use_keys`? if `key_file` is present then we can ignore `password`
Seems redundant to pass both `use_keys`and `key_file`

**Comment by srbharadwaj** [comment]
Also if u dont mind, give me a week's time i will commit this change along with some other changes when NXOS 9.2(1) gets released

**Comment by srbharadwaj** [comment]
Can pick the code from here and see if key_file is working as expected?
https://github.com/srbharadwaj/mdssdk?organization=srbharadwaj&organization=srbharadwaj

**Comment by wardy3** [comment]
Good point

I’ll add that and also remove the `isort` changes that snuck in without me noticing

Not sure how you fix that type of thing on a public repo. I normally `push -f` after a rebase. I’ll add new commits and I suppose we just squash them in this case?

From: Suhas Bharadwaj ***@***.***>
Date: Monday, 23 August 2021 at 20:28
To: Cisco-SAN/mdssdk ***@***.***>
Cc: Wardy ***@***.***>, Author ***@***.***>
Subject: Re: [Cisco-SAN/mdssdk] allow ssh key option (#11)

@srbharadwaj commented on this pull request.

________________________________

In mdssdk/switch.py<https://github.com/Cisco-SAN/mdssdk/pull/11#discussion_r693851858>:

> @@ -54,14 +55,17 @@ class Switch(SwitchUtils):



     :example:

         >>> switch_obj = Switch(ip_address = switch_ip, username = switch_username, password = switch_password)

+        >>> switch_obj = Switch(ip_address = switch_ip, username = switch_username, use_keys = True, key_file = './ssh/test_rsa')

Why do we need use_keys? if key_file is present then we can ignore password
Seems redundant to pass both use_keysand key_file

File: mdssdk/switch.py (hunk 1 above)
Comment (wardy3) [reply mdssdk/switch.py]: fixed...

**Comment by srbharadwaj** [comment]
@wardy3 Closing this PR as the latest release v1.3.0 should support this feature, please test it out and let me know

**Comment by wardy3** [comment]
> @wardy3 Closing this PR as the latest release v1.3.0 should support this feature, please test it out and let me know

looks good. thanks!

**Pull request closed without merging**